inf-ruby.el

```diff
@@ -282,6 +284,20 @@ The following commands are available:
       (inf-ruby-remove-in-string (buffer-substring (point) end)
                                  inf-ruby-prompt-pattern))))
 
+(defun inf-ruby-buffer ()
+  "Return inf-ruby buffer for the current buffer or project."
+  (catch 'buffer
+    (let ((current-dir         (file-truename default-directory))
+          (current-project-dir (file-truename
+                                (locate-dominating-file default-directory
+                                                        #'inf-ruby-console-match))))
```

Comment:
> That's a big change, what about the legacy code?

It's only a change in the set of interactive functions. The functions themselves, which could be used somewhere programmatically, will still be available. The semantics of the DIR argument will slightly change (should we rename it to ROOT?), but that's less of a breaking change.